Content/WebPage/WebPageViewController.swift

```diff
@@ -0,0 +1,155 @@
+//
+//  WebPageViewController.swift
+//  RealifeTech
+//
+//  Created by Mickey Lee on 02/03/2021.
+//  Copyright © 2021 Realife Tech. All rights reserved.
+//
+
+import UIKit
+import WebKit
+
+public protocol WebPageViewControllerDelegate: AnyObject {
+    func webViewCanGoBack()
+    func webViewCanGoForward()
+    func webViewEstimatedProgressChanged(_ progress: Double)
+    func urlForNavigationAction(_ url: URL?)
+    func webViewDidStartProvisionalNavigation()
+    func webViewDidFinishNavigation()
+}
+
+public class WebPageViewController: UIViewController {
+
+    let result: Result<WebPage, Error>
+    weak var delegate: WebPageViewControllerDelegate?
+
+    private(set) var canGoBackObserver: NSKeyValueObservation?
+    private(set) var canGoForwardObserver: NSKeyValueObservation?
+    private(set) var estimatedProgressObserver: NSKeyValueObservation?
+
+    public lazy var webView: WKWebView = {
```

Comment:
I set the access level of the webView to public for now as I'm not sure whether the apps need to access it by any chance. @SoaringEarth Any thoughts?

Reply:
That's fine- there's nothing mentioned, but it doesn't hurt to give apps the option